### PR TITLE
CRNS-282: RN 0.64, Expo 41 - upgrade packages and make corresponding changes

### PR DESCRIPTION
**Pull request by virdesai**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

## Description of the pull request
Mostly focused on upgrading
react-native: 0.63.4 -> 0.64.1
react-native-reanimated: 2.0.0-rc.0 -> 2.1.0
@gorhom/bottom-sheet: 3.0.0-alpha.0 -> 3.6.4
expo: 41.0.1
stream-chat: 3.9.0

### CODE REVIEW

**Comment by vishalnarkhede** [comment]
I will take a look in details tomorrow. :)

**Comment by vishalnarkhede** [comment]
It looks fine to me. Could you write some changelog @virdesai ? Which I can copy paste later when publishing

File: native-package/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "react-native-haptic-feedback": ">=1.11.0",
     "react-native-image-crop-picker": ">=0.33.2",
     "react-native-image-resizer": ">=1.4.2",
-    "@stream-io/flat-list-mvcp": "0.0.2",
+    "@stream-io/flat-list-mvcp": "0.0.9",
```
Comment (vishalnarkhede) [inline native-package/package.json:26]:
```suggestion
    "@stream-io/flat-list-mvcp": "0.10.1",
```

**Comment by giautm** [comment]
This PR scare me out. 🤧🤧🤧. Too many files changed.

**Review by vishalnarkhede — APPROVED** [review]

**Comment by nhannah** [comment]
> This PR scare me out. 🤧🤧🤧. Too many files changed.
> 
> Why not splitting into multi-PRs instead?

They are interdependent; we have had to wait for expo 41 to upgrade reanimated and RN 64; these had breaking changes which  we had to adjust for.

**Comment by giautm** [comment]
Hey @nhannah, I just rebase https://github.com/GetStream/stream-chat-react-native/pull/692 on this. Can you review it?

**Pull request merged**